developers/prerequisites/OSGi-tasks.md

```diff
@@ -0,0 +1,198 @@
+
+### V. Events
+
+1. In the ```org.openhab.training.util``` bundle create and register as a service ``` TimeEventSender```, that sends events with topic "org/openhab/training/time" every minute with the current time stamp (hh:mm) (you can use the ```TimeUtilities``` class form chapter [I.](#i-writing-basic-osgi-bundle)).
+
+     > Hint! Install ```org.eclipse.equinox.event``` bundle in order to use *EventAdmin* service for sending events.  
+     > Hint! [ScheduledExecutorService](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html) can be used in this task.
+
+2. Modify the ```org.openhab.training.electricity.tv.TV``` to listen for events with this topic and print an error on the console, when one event is not received.
+
+### VI. Managed Services
+
+1. Implement the [*ManagedService*][ManagedService] interface in the ```org.openhab.training.electricity.tv.TV``` class.
+2. Add a configuration to the ```TV``` class with property "autoSleep" and value a timestamp (hh:mm).
+3. Modify the method that listens for the events with topic "time" from section V.:
+	- when the "autoSleep" property is equal to the "time" topic, display message on the console, that ```TV``` goes to sleep mode and stop listening for events.
+4. Modify the ```Battery``` provider by implementing the *ManagedService* interface. After this modification, it should be possible to change the charge of a battery.
+
+### VII. Console Commands
+
+1. Implement a service in  the ```org.openhab.training.util``` bundle that adds console command, that:
+	- displays all  ```ElectricityProvider``` implementations that are registered with their charge;
+	- displays all consumers, to which providers are they connected (which providers are available) and what is the current provider at the moment;
+
+    > Hint! You might have to register all consumers as services in the OSGi ServiceRegistry.
+
+	- recharges the ```Battery``` with a provided value;
+	- changes the configuration of the ```TV```.
+
+    > Hint! You can add console commands with the help of
+[CommandProvider][CommandProvider]
+
+2. Test the console commands in the OSGi runtime by starting the different Radio consumers and the TV consumer. Start only the ```Battery``` provider and wait until it is discharged. Check if the consumers have stopped as expected. Recharge the battery and check, if the consumers have started. You can try out different scenarios as well!
+
+## References 
+
+Some of the tutorials listed below are not up-to-date with the latest versions of the Equinox framework or are written to be run in another implementation of the OSGi Runtime (e.g. Apache Felix). They can help you with solving the tasks above. This resources are divided in several groups.
+
+### Multiple topics
+{:.no_toc}
+- This tutorials simply cover more than one from the topics below:
```

Comment:
I'm on another PC and can not look at your article after it was processed by Jekyll. Did you look at the results?
Something like this will not be rendered correctly by Jekyll, because of a missing empty line between headline and content. Maybe `{:.no_toc}` mekes a difference here...?